visualization/pg_joins.js

```diff
@@ -52,40 +52,67 @@ const fetch_specieId= async (client, specie)=>{
   }
 }
 
+const getGeometry= async (client, geometry)=>{
+  try{
+    geometry.coordinates[0].map((data)=>{
+      if(data.length<3)
+        data.push(115);
+      return data
+    })
+    const res= (await client.query('SELECT ST_Geomfromtext(ST_AsText(ST_GeomFromGeoJSON($1)),4326) As wkt', [geometry]));
+    return res.rows[0]["wkt"]
+  }
+  catch(e){
+    console.log(e);
+  }
+}
 /*We just need the data after performing joins on some tables, for only visualization purposes not insertion,
  therefore a ROLLBACK is performed at end of this function*/
 
-const getCultivarSites=async (cultivars, sites_cultivars, sites) => {
+const getCultivarSites=async (cultivars, sites_cultivars, sites, geoJSON) => {
   const client = await pool.connect()
   try{
     await client.query('BEGIN')
+    let t1= ((await client.query("SELECT (UTC_NOW() - interval '1 second') as utc_now")).rows[0]["utc_now"]);
```

Comment:
Is this sufficient amount of time into the past? Perhaps getting the system time before making the queries, caching and using it for these queries (here and elsewhere that use the same time specification) would more certain to capture all the data?

Reply:
Yeah! I checked that and it is sufficient enough. Also, getting the system time before making queries is the same as the time taken while performing them so it is not making any difference.